lib/module.js

```diff
@@ -11,13 +11,14 @@ module.exports = function (moduleOptions) {
     injectModule: true
   }
 
-  if (this.nuxt.options.dev && !('legacyMode' in this.options.firebase) && !('legacyMode' in moduleOptions)) {
-    logger.info('A legacyMode has been introduced in v7 which will by default be false in v8. Set \'legacyMode: true\' in your config to remove this message or migrate to the new API. Read more here: XX')
-  }
-
   const options = Object.assign(defaultOptions, this.options.firebase, moduleOptions)
   const currentEnv = getCurrentEnv(options)
 
+  // TODO: Remove with v8
+  if (this.nuxt.options.dev && !(this.options.firebase && 'legacyMode' in this.options.firebase) && !('legacyMode' in moduleOptions)) {
```

Comment:
didn't really mean to move it down, it's rather about the `this.options.firebase` that can be undefined